tests/api.rs

```diff
@@ -0,0 +1,83 @@
+mod support;
+
+#[cfg(feature = "api")]
+mod tests {
+    use crate::support::{sink, source};
+    use reqwest::Response;
+    use std::net::SocketAddr;
+    use std::time::Duration;
+    use vector::api;
+    use vector::config::Config;
+    use vector::test_util::{next_addr, retry_until};
+
+    // provides a config that enables the API server, assigned to a random port. Implicitly
+    // tests that the config shape matches expectations
+    fn api_enabled_config() -> Config {
+        let mut config = Config::builder();
+        config.add_source("in1", source().1);
+        config.add_sink("out1", &["in1"], sink(10).1);
+        config.api.enabled = true;
+        config.api.bind = Some(next_addr());
+
+        config.build().unwrap()
+    }
+
+    // spawns the API server on the provided port; optionally enables playground
+    fn start_api_server(addr: SocketAddr, playground: bool) {
+        tokio::spawn(async move {
+            let _ = api::make_server(addr, playground).await;
```

Comment:
This will be spawned independently for each test. Could drop this at the end of each test but not sure it's worth explicitly handling, since these tests run for all of 300ms.

Reply:
Our entire test suite can take some time though, and having active API servers could definitely impact testing. We run our tests on fairly slow CI boxes, so it's probably safer for reliability reasons to properly reap this.

Reply:
That's a good point. I'll make these tear-downable 👍

Reply:
Updated the API implementation to allow for `server.stop()`, which shuts down the API gracefully. Updated tests to use it in 80bd6615e0290a75855a22bae2e1d404f0e2f8eb

Reply:
Argh, no I didn't. It's actually in https://github.com/timberio/vector/pull/3778/commits/9e1a79b40ec4214292c8e9eaea86620f9f495893 -- the GraphQL Client branch. I'll bring those changes over.